dev-docs/Action Flows.md

```diff
@@ -0,0 +1,349 @@
+
+## Fetching / Updating Verifiable Credential (VC)
+
+Once a VC has been issued under the Public ID of a particular user, it needs to be fetched. Previously fetched VCs also need to be checked periodically for possible updates.
+
+The two possible data sources via which this can be done has been described towards the end of the [Issuing a new Verifiable Credential (VC)](#issuing-a-new-verifiable-credential-vc) section.
+
+Once the VC has been fetched, the credential's signature is verified and it is ensured that it has been issued by the genuine issuer. For more details on how signatures are verified, you can check out the [Verification of a VC or VP](#verification-of-a-vc-or-vp) section.
+
+Once the signature is verified, the JSON data is fetched and written to the local storage of the user's device where it persists for future use and creation of Verifiable Presentations.
+
+## Verification of a VC or VP
+
+The proof section of a VC or VP shall also contain the public key or URL to the public key to be used in order to verify the authenticity of the signature present in the JSON object.
+
+A verifier node simply caches the public keys which it receives during verification requests. It then uses this public key along with the signature passed in the request to verify that the signature was indeed signed by the right signer (issuer).
+
+After verification it then simply returns the verification status of the Credential / Presentation which could be one of the following
+
+- **Verified Document :** The public key was successfully fetched and used to verify that the signer was indeed the issuer itself.
+
+- **Unable to Verify Document :** In case the public key was given as a URL, and could not be fetched or decoded properly, it would be unclear as to whether the document is actual authentic or not. Hence this status is returned when the system is unable to determine the authenticity of a VC or VP.
+
+- **Tampered Document :** In case the public key was successfully fetched but the signature is not authentic.
+
+## Requesting for a Verifiable Presentation (VP)
+
+Once VCs have been issued, fetched, verified and added to the competency passbook of a holder, there needs to be a way for other 3rd parties to access / request for this specific data in a safe and secure manner.
+
+Let’s suppose a company would like to extend an internship offer to a student. In order to create the offer letter and register the student in their database they require specific details from multiple documents. For example: Aadhar Card Number, PAN Card Number, etc
+
+The Requester, looks for the official and verified JSON Aadhar VC template in the template store. This allows the requester to know which type of data is available with the holder of the VC. The requester then creates a request containing the specific fields from specific VCs which they need access to. This request is then signed using their private key.
```

Comment:
I would imagine we would need to create a standardised set of minimal attributes as part of the specification, so that requesters are able to request for those, instead of having to know which "document" contains which attributes. Let's branch that off to a separate PR containing a v0